sycl/test-e2e/Matrix/joint_matrix_bf16_fill_k_cache_impl.hpp

```diff
@@ -245,24 +327,30 @@ void test() {
   queue q;
   T *A = malloc_shared<T>(MATRIX_SIZE * MATRIX_SIZE, q);
   T *B = malloc_shared<T>(MATRIX_SIZE * MATRIX_SIZE, q);
+#ifdef VNNI
   T *vnniB = malloc_shared<T>(MATRIX_SIZE * MATRIX_SIZE, q);
+#endif
   TResult *C = malloc_shared<TResult>(MATRIX_SIZE * MATRIX_SIZE, q);
   TResult *refC = malloc_shared<TResult>(MATRIX_SIZE * MATRIX_SIZE, q);
 
   matrix_rand<T>(MATRIX_SIZE, MATRIX_SIZE, A, T(1));
   matrix_rand<T>(MATRIX_SIZE, MATRIX_SIZE, B, T(1));
-  matrix_vnni<T>(MATRIX_SIZE, MATRIX_SIZE, B, vnniB, VNNI);
 
   matrix_multiply_ref<T, T, TResult, 1>(A, B, refC, MATRIX_SIZE, MATRIX_SIZE,
                                         MATRIX_SIZE);
 
+#ifdef VNNI
+  matrix_vnni<T>(MATRIX_SIZE, MATRIX_SIZE, B, vnniB, vnniFactor);
+  B = vnniB;
```

Comment:
Because of B pointer was rewritten, memory allocated by this `T *B = malloc_shared<T>(MATRIX_SIZE * MATRIX_SIZE, q);` is not freed.
Either do not rewrite the pointer and use vnniB and free it in the end (my preference, as it seems less code), or save B pointer to some temporary, before rewriting, so you can free the memory in the end.